### DISCUSSION TITLE
Does the class bis.spatialCategory has  any relationship with “family” in revit

### DISCUSSION THREAD

**Discussion by YeMengHua417** [iTwin/itwinjs-core#4735] [Q&A]
1 Does the class bis.spatialCategory has any relationship with the “Family” in revit？we have found near half of the categories overlap with the “Family” in revit, but I have checked the definition from spatialCategory, which seems the category for 3d elements,
2 Is there any way to get all the categoris define in revit “family” from Bis schema? 
![image](https://user-images.githubusercontent.com/22695129/204734738-f601942d-bf67-44e8-af2b-1c40736244bd.png)

**Comment by ColinKerr** [comment]
Geometric Elements belong to a category, a bis.SpatialCategory is a specialization specifically for 3d elements.  The documentation for [category](https://www.itwinjs.org/bis/guide/fundamentals/categories/) explains this in more detail.

I believe Revit families will get converted into bis classes deriving from [bis.PhysicalElement](https://www.itwinjs.org/bis/guide/fundamentals/element-fundamentals/#geometricelement).

If you want to visually see the classes that were created you can use the iModel Schema Explorer to view the schemas in a particular iModel.  You can construct the URL like this: `https://imodelschemaeditor.bentley.com/?itwinid=<iTwin Id>&imodelid=<iModel Id>&changesetid=<ChangeSet Id>`.  

Or you can open the iModel with the [iModel Console](https://imodelconsole.bentley.com/) then select the 'Schema Explorer' option from the gear drop down:
![image](https://user-images.githubusercontent.com/33028649/204863020-f5f3d647-7eb7-49f3-821a-9388598d3a6d.png)

> NOTE: Once you have opened your iModel in with the iModel Console you can query for all spatial categories using this query `SELECT CodeValue, UserLabel FROM bis.SpatialCategory `, this will select the the two properties which may store the name of the category shown in the UI.

In the iModel Schema Explorer search for one of the faimlies and pick the class with the matching name.  The tree should show the class hierarchy and the pane on the right hand side should show details about the class.  The classes properties should be similar to the properties you see in Revit but will include additional properties coming from base BIS classes.

Let me know if I can clarify anything or if I can go into anymore details on any subject
